### PR TITLE
Various Fixes

### PR DESCRIPTION
**Pull request by varunagrawal**
A set of small fixes that have crept up over time.

### CODE REVIEW

**Comment by varunagrawal** [comment]
The fix for the CI failure is already in #1501. Will merge `develop` into this once that lands to resolve the CI.

**Comment by dellaert** [comment]
gotcha

**Review by dellaert — APPROVED** [review]

**Pull request merged**
